controllers/storagecluster/cephcluster.go

```diff
@@ -766,6 +767,35 @@ func getMonCount(sc *ocsv1.StorageCluster) int {
 	return defaults.DefaultMonCount
 }
 
+// getFullRatios returns the ratio for the given ratio name
+// If the ratio is not set in the StorageCluster CR, it returns the default value
+func getFullRatios(sc *ocsv1.StorageCluster, ratioName string) *float64 {
```

Comment:
Can we change the function signature to
`func getFullRatios(sc *ocsv1.StorageCluster, ratioName string, returnDefaultIfNotSet bool) *float64 {`
and change the call places with apt 'returnDefaultIfNotSet' knob.

Tried to re-imagine the function as below, please take a look,
```go
// getFullRatios returns the ratio for the given ratio name
// If the ratio is not set in the StorageCluster CR,
// it returns the default value if the 'returnDefaultIfNotSet' is true otherwise 'nil'
func getFullRatios(sc *ocsv1.StorageCluster, ratioName string, returnDefaultIfNotSet bool) *float64 {
	const (
		NearFull     = "nearFull"
		Full         = "full"
		BackfillFull = "backfillFull"
	)
	// always safe to use a 'Copy' as we are returning pointers
	mCCSpec := sc.Spec.ManagedResources.CephCluster.DeepCopy()
	mapRatioWithName := map[string]*float64{
		NearFull:     mCCSpec.NearFullRatio,
		Full:         mCCSpec.FullRatio,
		BackfillFull: mCCSpec.BackfillFullRatio,
	}
	if returnDefaultIfNotSet {
		if mapRatioWithName[NearFull] == nil {
			mapRatioWithName[NearFull] = ptr.To(defaults.DefaultNearFullRatio)
		}
		if mapRatioWithName[Full] == nil {
			mapRatioWithName[Full] = ptr.To(defaults.DefaultFullRatio)
		}
		if mapRatioWithName[BackfillFull] == nil {
			mapRatioWithName[BackfillFull] = ptr.To(defaults.DefaultBackfillFullRatio)
		}
	}
	// for any other unintended 'ratioName', the function will return 'nil'
	return mapRatioWithName[ratioName]
}
```

Reply:
@aruniiird Thanks for the detailed function code & the suggestion but I have updated the logic to not use this function at all.